### PR TITLE
fix(rn-example): update metro config and deps

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add react-core-notifs dep to RN Example App and update internal dependency addition docs in _README_

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/react-native/metro.config.js [hunk 1]
```diff
@@ -9,7 +9,12 @@ const INTERNAL_DEPENDENCY_FLAG = '*';
 const INTERNAL_DIRECTORY_ROOTS = ['packages'];
 
 // internal packages used by the RN example
-const INTERNAL_USED_PACKAGES = ['ui', 'react-core', 'react-native'];
+const INTERNAL_DEPENDENCY_DIRECTORY_NAMES = [
```
Comment (calebpollman) [inline examples/react-native/metro.config.js:12]: Decided to update the namespace here since the value here and dependency names are not 1:1

**Review by ioanabrooks — APPROVED** [review]
Thanks for fixing this! The example app still doesn't work, environments imports need to be fixed post split pkg work. I have a PR out for it: https://github.com/aws-amplify/amplify-ui/pull/3865

**Comment by calebpollman** [comment]
> Thanks for fixing this! The example app still doesn't work, environments imports need to be fixed post split pkg work. I have a PR out for it: #3865

Thanks for opening that!

**Review by zchenwei — APPROVED** [review]
LGTM

**Review by reesscot — APPROVED** [review]

**Pull request merged**